### ISSUE TITLE
[SOLVED] Impossible to add item in migx tab from migx management...

### ISSUE THREAD

**Issue by nfl069** [Bruno17/MIGX#227]
Hi !

I can't save/add item in migx tab from migx management. When i click on the "Add Item" button, i have "MIGX" pop-up. I can fill the fields with package's parameters. But when i click on the "done" button, i just have a loading pop-up and nothing else : Registration is not done !

I do not have any error message at screen like in log...

I had the same behavior with this MODX issue : #12580

https://github.com/modxcms/revolution/issues?utf8=%E2%9C%93&q=12580

But now it's patched since MODX Revo 2.4.2-pl and all works fine.

Do you have any ideas, please?

MODX Revo 2.4.2-pl (Traditional) [Updated from 2.4.1 to 2.4.2]
MIGX 2.9.6-pl + patch for issue #222

**Comment by cjac-itafran** [comment]
Hi!
Same problem

I can't save item

MODX Revo 2.4.2-pl (Traditional)
MIGX 2.9.6-pl + patch for issue #222

**Comment by nfl069** [comment]
Hi Bruno !

To reproduce the problem you need to set the debug key's value to 4 in the modx system settings (and specifically activate E_NOTICE error from PHP).

Now when you go to MIGX Management, if you look at your javascript console (i use firefox) you have an error like this : SyntaxError: expected expression, got '<'

In the Network tab, look at the XHR request 's response file connector.php. You have HTML tags before JSON. HTML tags are E_NOTICE error from PHP : 

Undefined variable: packageName in core/components/migx/processors/mgr/default/getlist.php on line 51

I think error is before line 51...?

Thank to Bob RAY for his article : http://bobsguides.com/blog.html/2013/05/22/debugging-cmps-a-horror-story/

Of course if you don't activate E_NOTICE error from PHP, you have no error!

Hope this can help you to correct this bug?

**Comment by Bruno17** [comment]
does this help?
https://github.com/Bruno17/MIGX/commit/3a58fddc692d61b36f428e4b1bfabf53bee2282d

**Comment by nfl069** [comment]
It works much better! A great big thank you!

But however I still have a little problem. In the Network tab, look at the XHR request 's response file connector.php. I've an error with JSON : 

SyntaxError: JSON.parse: unexpected﻿ character at line 1 column 1 of the JSON data

There is a strange character (like a dot) before {"success":true,"total":"1","results"... 

I don't know why? Could you verify please?

Thanks in advance

![json_character_error](https://cloud.githubusercontent.com/assets/8918088/10365150/e0f6309a-6db8-11e5-99e1-57e04691f3b1.jpg)

**Comment by nfl069** [comment]
Hi Bruno !

Supplement...

 - BEFORE PATCH & with unactivated E_NOTICE : 

![json_ok](https://cloud.githubusercontent.com/assets/8918088/10388050/94cac7fc-6e5e-11e5-9876-38e652649a1e.jpg)

 - AFTER PATCH & with unactivated E_NOTICE : 

![json_character_error](https://cloud.githubusercontent.com/assets/8918088/10388063/b005105e-6e5e-11e5-83fd-9384067950c5.jpg)

Hope this can help you to correct this bug?

Thanks in advance

**Comment by Bruno17** [comment]
can't reproduce that issue. Maybe you did something wrong when patiching the files.
Or did you download the modified files?

**Comment by nfl069** [comment]
I found...

This was a problem with UTF-8 encoding with BOM (Byte Order Mark)/zero-width no-break space. I downloaded the .zip file from MIGX github instead of RAW copy/paste.

Just for information, XHR request's Content-Type is :"application/json; charset=UTF-8".

Now everything works!

It would be well to have a new release of MIGX...?

So i could completely remove MIGX 2.9.6-pl and then install a new MIGX 2.9.(7?)-pl release directly from etxras management menu in ModX.

Thank you for your help Bruno!
